packages/expo-screen-orientation/ios/ScreenOrientationRegistry.swift

```diff
@@ -0,0 +1,239 @@
+import Foundation
+import ExpoModulesCore
+protocol OrientationListener {
+  func screenOrientationDidChange(_ orientation: UIInterfaceOrientation)
+}
+
+// This singleton holds information about desired orientation for every app which uses expo-screen-orientation.
+// Marked @objc and public, because this it is also used in EXAppViewController.
+@objc
+public class ScreenOrientationRegistry: NSObject, UIApplicationDelegate {
+  @objc
+  public static let shared = ScreenOrientationRegistry()
+
+  var currentScreenOrientation: UIInterfaceOrientation
+  var orientationListeners: [ScreenOrientationModule?] = []
+  var moduleInterfaceMasks: [ScreenOrientationModule: UIInterfaceOrientationMask] = [:]
+  weak var currentTraitCollection: UITraitCollection?
+  var lastOrientationMask: UIInterfaceOrientationMask
+  var rootViewController: UIViewController? {
+    return UIApplication.shared.windows.filter { $0.isKeyWindow }.first?.rootViewController
+  }
+
+  var currentOrientationMask: UIInterfaceOrientationMask {
+    var currentOrientationMask: UIInterfaceOrientationMask = []
+
+    EXUtilities.performSynchronously {
+      currentOrientationMask = self.rootViewController?.supportedInterfaceOrientations ?? []
+    }
+    return currentOrientationMask
+  }
+
+  private override init() {
+    self.currentScreenOrientation = .unknown
+    self.currentTraitCollection = nil
+    self.lastOrientationMask = []
+
+    super.init()
+
+    NotificationCenter.default.addObserver(
+      self,
+      selector: #selector(self.handleDeviceOrientationChange(notification:)),
+      name: UIDevice.orientationDidChangeNotification,
+      object: UIDevice.current
+    )
+
+    DispatchQueue.main.async {
+      UIDevice.current.beginGeneratingDeviceOrientationNotifications()
+    }
+  }
+
+  // Called by ScreenOrientationAppDelegate in order to set initial interface orientation
+  func updateCurrentScreenOrientation() {
+    let windows = UIApplication.shared.windows
+    if !windows.isEmpty {
+      self.currentScreenOrientation = windows[0].windowScene?.interfaceOrientation ?? .unknown
+    }
+  }
+
+  deinit {
+    EXUtilities.performSynchronously {
+      UIDevice.current.endGeneratingDeviceOrientationNotifications()
+    }
+  }
+
+  // MARK: affecting screen orientation
```

Comment:
`-` makes the difference in Xcode's method navigator. We used to use the line separator.
```suggestion
  // MARK: - Affecting screen orientation
```